### PR TITLE
can-define/list inheritable

### PR DESCRIPTION
**Pull request by cherifGsoul**
@justinbmeyer can you check this if we can add more test use cases

### CODE REVIEW

**Comment by justinbmeyer** [comment]
This is great.  The only other thing I would probably test for with list is making sure `"*"` inheritance works.

```js
BaseList = DefineList.extend({
  "*": SomeMapType
});

ExtendedList = BaseList.extend({});

var xl = new ExtendedList([{}]);

QUnit.ok(xl[0] instanceof SomeMapType);

```

**Comment by cherifGsoul** [comment]
@justinbmeyer it is done, can you check and merge please.

File: list/list-test.js [hunk 1]
```diff
@@ -545,3 +545,104 @@ QUnit.test("is list like", function(){
...
+		amount: "number",
+		slug: {
+			serialize: true,
+			get: function(){
+				return this.name.toLowerCase().replace(/ /g,'-').replace(/[^\w-]+/g,'');
+			}
+		}
+	});
+
+  var BaseList = DefineList.extend({
```
Comment (justinbmeyer) [inline list/list-test.js]: This looks good.  But can you make sure tabs are used for indentation instead of spaces?  Thanks!

**Comment by cherifGsoul** [comment]
@justinbmeyer atom has a weired indentation configuration :smile:

**Comment by m-mujica** [comment]
@cherifGsoul I'd suggest you try http://editorconfig.org/, some of our projects have the `.editorconfig` file already in place, e.g https://github.com/stealjs/steal-tools/blob/master/.editorconfig

We can add that file to can-define, and you only need to install the editor plugin, then you don't need to worry about indentation settings :D

**Comment by cherifGsoul** [comment]
@m-mujica awesome thank you :+1:

**Pull request merged**